oval_graph/_xml_parser_test_info.py

```diff
@@ -107,18 +107,37 @@ def _xml_element_to_dict(self, object_, collected_object):
         else:
             result[object_.attrib.get('id')] = "does not exist"
             result[self._get_unique_id_in_dict(
-                object_, result)] = self._get_object_items(object_)
+                object_, result)] = self._get_object_items(object_, collected_object)
         return result
 
-    def _get_object_items(self, object_):
+    def _get_object_items(self, object_, collected_object):
         out = {}
         for element in object_.iterchildren():
             if element.text and element.text.strip():
                 out[self._get_unique_id_in_dict(element, out)] = element.text
             else:
-                out[self._get_unique_id_in_dict(element, out)] = "no value"
+                if self._collected_object_is_not_none_and_not_empty_and_element_contain_var_ref(
```

Comment:
Maybe you could move the whole else branch to a separate function. That can make the code more readable.